tests/rest/admin/test_room.py

```diff
@@ -351,6 +368,42 @@ def test_purge_room_and_not_block(self):
         self._is_blocked(self.room_id, expect=False)
         self._has_no_members(self.room_id)
 
+    def test_block_room_and_not_purge(self):
+        """Test to block a room without purging it.
+        Members will not be moved to a new room and will not receive a message.
+        The room will not be purged.
+        """
+        # Test that room is not purged
+        with self.assertRaises(AssertionError):
+            self._is_purged(self.room_id)
+
+        # Test that room is not blocked
+        self._is_blocked(self.room_id, expect=False)
+
+        # Assert one user in room
+        self._is_member(room_id=self.room_id, user_id=self.other_user)
+
+        body = json.dumps({"block": False, "purge": False})
```

Comment:
Is this a typo? Should this have been `"block": True`?

Reply:
Good catch. I think so.
Neither block nor purge is certainly not a good use case.
My PR #11223 needs an update, too.

Reply:
> Neither block nor purge is certainly not a good use case.

To check my understanding, I think no block and no purge would
- kick all local users out of the room
- retain events for that room
- all users would be free rejoin

which I agree, sounds like it ought to caught and rejected with 400 bad request

Reply:
I don't think you have to check everything. Maybe it's a edge case.

Reply:
My concern is that someone might accidentally use the API in this way, and not realise that the room can come back to haunt them.

(Is there a use case for shutting down the room with `block: False` at all?)

Reply:
I have no idea for an use case `{"block": False, "purge": False}` and remove all users.